Directory.Build.props

```diff
@@ -0,0 +1,102 @@
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <PackageProjectUrl>$(RepositoryUrl)</PackageProjectUrl>
+    <ProduceReferenceAssembly>true</ProduceReferenceAssembly>
+    <RepositoryType>git</RepositoryType>
+    <RestoreSources>
+      https://www.myget.org/F/sixlabors/api/v3/index.json;
+      https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json;
+      https://api.nuget.org/v3/index.json;
+    </RestoreSources>
+    <SixLaborsPublicKey>002400000c8000009400000006020000002400005253413100040000010001000147e6fe6766715eec6cfed61f1e7dcdbf69748a3e355c67e9d8dfd953acab1d5e012ba34b23308166fdc61ee1d0390d5f36d814a6091dd4b5ed9eda5a26afced924c683b4bfb4b3d64b0586a57eff9f02b1f84e3cb0ddd518bd1697f2c84dcbb97eb8bb5c7801be12112ed0ec86db934b0e9a5171e6bb1384b6d2f7d54dfa97</SixLaborsPublicKey>
+    <UseSharedCompilation>true</UseSharedCompilation>
+  </PropertyGroup>
+
+  <!-- Package versions for package references across all projects -->
+  <PropertyGroup>
+    <BenchmarkDotNetPackageVersion>0.11.5</BenchmarkDotNetPackageVersion>
+    <ColourfulPackageVersion>2.0.2</ColourfulPackageVersion>
+    <DotnetXUnitPackageVersion>2.3.1</DotnetXUnitPackageVersion>
+    <MagicNetQ16AnyCpuPackageVersion>7.12.0</MagicNetQ16AnyCpuPackageVersion>
+    <MicrosoftNetCompilersToolsetPackageVersion>3.1.0-beta2-final</MicrosoftNetCompilersToolsetPackageVersion>
+    <MicrosoftNetTestSdkPackageVersion>15.9.0</MicrosoftNetTestSdkPackageVersion>
+    <MoqPackageVersion>4.10.0</MoqPackageVersion>
+    <SixLaborsCorePackageVersion>1.0.0-dev000101</SixLaborsCorePackageVersion>
+    <SixLaborsFontsPackageVersion>1.0.0-beta0008</SixLaborsFontsPackageVersion>
+    <SixLaborsShapesTextPackageVersion>1.0.0-beta0007</SixLaborsShapesTextPackageVersion>
+    <SixLaborsShapesPackageVersion>1.0.0-beta0008</SixLaborsShapesPackageVersion>
+    <StyleCopAnalyzersPackageVersion>1.1.118</StyleCopAnalyzersPackageVersion>
+    <SystemDrawingCommonPackageVersion>4.5.1</SystemDrawingCommonPackageVersion>
+    <SystemIOCompressionPackageVersion>4.3.0</SystemIOCompressionPackageVersion>
+    <SystemIOUnmanagedMemoryStreamPackageVersion>4.3.0</SystemIOUnmanagedMemoryStreamPackageVersion>
+    <SystemRuntimeCompilerServicesUnsafePackageVersion>4.5.1</SystemRuntimeCompilerServicesUnsafePackageVersion>
+    <SystemThreadingTasksParallelPackageVersion>4.3.0</SystemThreadingTasksParallelPackageVersion>
+    <SystemValueTuplePackageVersion>4.5.0</SystemValueTuplePackageVersion>
+    <XUnitPackageVersion>2.3.1</XUnitPackageVersion>
+    <XUnitRunnerConsolePackageVersion>2.3.1</XUnitRunnerConsolePackageVersion>
+    <XUnitRunnerVisualStudioPackageVersion>2.3.1</XUnitRunnerVisualStudioPackageVersion>
+  </PropertyGroup>
+
+  <!-- Package references which are consumed by all projects -->
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="$(MicrosoftNetCompilersToolsetPackageVersion)" IsImplicitlyDefined="true" />
```

Comment:
This is the Roslyn compiler toolset and ensures all machines build with the same compiler version, regardless of what is on the box.

`IsImplicitlyDefined=true` just means users can't remove this from the list of NuGet packages in the UX.

Reply:
TIL